src/services/ThirdPartyService.ts

```diff
@@ -34,10 +44,77 @@ export class ThirdPartyService implements IThirdPartyService {
             });
 
             this.logger.verbose(`Verified GPIB for user ${userInfo.userId}`);
+
+            //Save the verify request
+            await this.requestRepository.save({
+                from: "IDEM",
+                to: VenderEnum.GPIB,
+                ipAddress: ip,
+                requestType: RequestType.Verify
+            });
             return true;
         } catch (error) {
             this.logger.error(error.message);
             throw new Error(error);
         }
     }
+
+    public async signup(
+        signupInfo: UserSignupRequest,
+        ip: string
+    ): Promise<string> {
+        let requestBody = {};
+        let endPoint: string;
+
+        if (signupInfo.source === VenderEnum.GPIB) {
+            endPoint = this.config.get(ConfigSettings.GPIB_SIGNUP_ENDPOINT);
+            requestBody = {
+                firstName: signupInfo?.firstName,
+                lastName: signupInfo?.lastName,
+                email: signupInfo.email,
+                password: signupInfo.password,
+                referralCode: "",
+                trackAddress: true,
+                createAddress: true
+            };
+        } else if (signupInfo.source === VenderEnum.CoinStash) {
+            endPoint = this.config.get(
+                ConfigSettings.COINSTASH_SIGNUP_ENDPOINT
+            );
+            requestBody = {
+                email: signupInfo.email,
+                password: signupInfo.password,
+                displayName: `${signupInfo?.firstName} ${signupInfo?.lastName}`,
+                country: "Australia",
+                token: this.config.get(ConfigSettings.COINSTASH_TOKEN),
+                acceptMarketing: true
```

Comment:
Maybe this should be false?

```suggestion
                acceptMarketing: false
```